### PR TITLE
IC13/IC14 Generate SPG Unit Addressed PQs

### PR DESCRIPTION
**Pull request by LeoKHoward**
# Motivation and Context
Want to generate individual addressed initial contact questionnaires for SPG Units in English and Welsh

# What has changed
New ActionTypes/PackCodes included
New integration and unit tests

# How to test?
Build and run with other repos and then run ATs

# Links
https://trello.com/c/nxTRcBaY/808-ic13-ic14-generate-spg-addressed-ic-pq-print-files-13

### CODE REVIEW

**Comment by LeoKHoward** [comment]
[(IC13/IC14) Generate SPG Addressed IC PQ Print Files (13)](https://trello.com/c/nxTRcBaY/808-ic13-ic14-generate-spg-addressed-ic-pq-print-files-13)

File: src/main/java/uk/gov/ons/census/action/builders/QidUacBuilder.java [hunk 1]
```diff
@@ -83,11 +86,11 @@ private UacQidTuple fetchExistingUacQidPairsForAction(Case linkedCase, ActionTyp
       throw new RuntimeException(
           String.format("We can't process this case id '%s' with no UACs", caseId));
 
-    } else if (!actionType.equals(ActionType.ICHHQW)
+    } else if ((!actionType.equals(ActionType.ICHHQW) && !actionType.equals(ActionType.SPG_IC14))
```
Comment (NickGrantX) [inline src/main/java/uk/gov/ons/census/action/builders/QidUacBuilder.java]: I would rewrite this whole chunk like this:

```
    if (uacQidLinks == null || uacQidLinks.isEmpty()) {
      throw new RuntimeException(
          String.format("We can't process this case id '%s' with no UACs", caseId));

    } else if (actionType == ActionType.ICHHQW || actionType == ActionType.SPG_IC14) {
      if (isStateCorrectForSecondWelshUacQidPair(linkedCase, uacQidLinks)) {
        return getUacQidTupleWithSecondWelshPair(uacQidLinks);
      }
    } else if (isStateCorrectForSingleUacQidPair(linkedCase, uacQidLinks)) {
      return getUacQidTupleWithSinglePair(uacQidLinks);
    }
    
    throw new RuntimeException(
        String.format("Wrong number of UACs for treatment code '%s'", actionType));
```

**Review by NickGrantX — CHANGES_REQUESTED** [review]
See comment

File: src/main/java/uk/gov/ons/census/action/builders/QidUacBuilder.java (hunk 1 above)
Comment (LeoKHoward) [reply src/main/java/uk/gov/ons/census/action/builders/QidUacBuilder.java]: Ok but why have you removed the second parameter from **_getUacQidTupleWithSecondWelshPair_**?

Comment (NickGrantX) [reply src/main/java/uk/gov/ons/census/action/builders/QidUacBuilder.java]: that was a mistake

Comment (LeoKHoward) [reply src/main/java/uk/gov/ons/census/action/builders/QidUacBuilder.java]: Ah ok...just checking

**Review by ryangrundy7 — APPROVED** [review]
LGTM 👍

**Review by NickGrantX — APPROVED** [review]
LGTM 👍

**Pull request merged**